clippy_lints/src/methods/mod.rs

```diff
@@ -4041,6 +4042,29 @@ declare_clippy_lint! {
     "calling `.get().is_some()` or `.get().is_none()` instead of `.contains()` or `.contains_key()`"
 }
 
+declare_clippy_lint! {
+    /// ### What it does
+    /// It identifies calls to `.is_empty()` on a value initialized from a literal string.
+    ///
+    /// ### Why is this bad?
+    /// Literal strings are known at compile time. Their emptyness can be determined statically.
```

Comment:
```suggestion
    /// String literals and constant values are known at compile time. Checking if they
    /// are empty will always return the same value. This might not be the intention of
    /// the expression.
```

Reply:
Done.